### PR TITLE
Add dtoa, ftoa, atod, atof functions

### PR DESCRIPTION
**Pull request by sclaiborne**
## What

Adds the following functions: `stringdtoa`, `stringftoa`, `stringatod`, `stringatof`. There functions are used to convert floating point numbers from/to string representation. 

## Why 

1. BR does not expose a dtoa or atod function. printf can be used as an alternative but is not thread safe. 
2. These new functions are thread safe

### CODE REVIEW

File: src/Ar/StringExt/stringdtoa.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+unsigned long stringftoa(float value, plcstring* buffer, unsigned long ndig, unsigned long bufferSize) {	
+	return g_ffmt(buffer, &value, ndig, bufferSize);
+}
+
+double stringatod(plcstring* value, plcstring** pEnd) {
+	return strtod(value, pEnd);
+}
+float stringatof(plcstring* value, plcstring** pEnd) {
+	return strtof(value, pEnd);
+}
```
Comment (lawrencegsullivan) [inline src/Ar/StringExt/stringdtoa.c:37]: I'm not sure I understand how pEnd is supposed to be used. Naively, It feels like there may be a friendlier api we could implement in this layer. Perhaps a string length or size being passed into stringatof/d?

**Review by lawrencegsullivan — COMMENTED** [review]
Fabulous. I have one question about the atof/d api.

File: src/Ar/StringExt/stringdtoa.c (hunk 1 above)
Comment (sclaiborne) [reply src/Ar/StringExt/stringdtoa.c:37]: Fair point. pEnd will be populated with a pointer to the end of the number. So its an output. We could instead return a length something like:
```c
float stringatof(plcstring* value, UDINT* pLen) {
	char* end = 0;
	float val = strtof(value, end);
	if(pEnd) *pEnd = (value-end);
	return val;
}
````

Comment (lawrencegsullivan) [reply src/Ar/StringExt/stringdtoa.c:37]: From linux man page:
> double atof(const char *nptr);
> 
> Description
> 
> The atof() function converts the initial portion of the string pointed to by nptr to double. The behavior is the same as
> 
>     strtod(nptr, NULL); 
> except that atof() does not detect errors.

It seems this implementation is closer to strtof/d than atof/d.
I think we could either keep the pEnd argument and update the name to indicate we are implementing strtof/d, or we could throw out the error checking and have the wrapper take only the value argument like standard atof/d functions.

In fact, I might be in favor of implementing both here - stringatof/d and stringstrtof/d
One provides error feedback via pEnd and the other is a slicker api

Comment (sclaiborne) [reply src/Ar/StringExt/stringdtoa.c]: I renamed them to stringstrtof/d. stringatof/d can be added in the future

File: example/AsProject/Logical/Programs/Default/Main.c
Comment (sclaiborne) [inline example/AsProject/Logical/Programs/Default/Main.c:1]: Reset some of this file

**Review by lawrencegsullivan — APPROVED** [review]

**Pull request merged**